internal/venice-common/src/main/java/com/linkedin/venice/utils/Utils.java

```diff
@@ -916,4 +916,14 @@ public static String getReplicaId(PubSubTopic topic, int partition) {
   public static String escapeFilePathComponent(final String component) {
     return component.replaceAll("[^a-zA-Z0-9-_/\\.]", "_");
   }
+
+  public static String getTaskStackTrace(String threadName) {
+    Thread thread = Thread.getAllStackTraces()
```

Comment:
This `getAllStackTraces()` function looks quite expensive... I assume we can probably do something better... For example, we could extend `DaemonThreadFactory` to keep track of created threads. See an example implementation here:

https://gist.github.com/FelixGV/b7ab92e97eebafdc2012c7508dbc9e71

Such an approach would also take care of the fragility concern in my other comment...

Reply:
Thanks Felix for the suggestion and code example. That's indeed a less expensive implementation and I would like to merge it into this change. One question for the example code is that could we have the same `threadNumber` in `RandomAccessDaemonThreadFactory` as in `DaemonThreadFactory`, so that it is 0-indexed, the same as `threadNumber` in DaemonThreadFactory, and `taskId` in ConsumptionTask.They are all all zero-indexed.

`threadNumber = super.threadNumber.get() + 1;` <--- how about we don't + 1 ?

Reply:
Oh yeah totally. I had assumed the parent class did `incrementAndGet()`, but actually it does `getAndIncrement()`, so then no +1 is needed in the subclass.

Reply:
I updated the gist. Feel free to steal this code, or adapt it any way you want.